addon/metrics-adapters/segment.js

```diff
@@ -0,0 +1,74 @@
+import Ember from 'ember';
+import canUseDOM from '../utils/can-use-dom';
+import { compact } from '../utils/object-transforms';
+import BaseAdapter from './base';
+
+const {
+  $,
+  assert,
+  copy,
+  get
+} = Ember;
+
+export default BaseAdapter.extend({
+  toStringExtension() {
+    return 'Segment';
+  },
+
+  init() {
+    const config = copy(get(this, 'config'));
+    const segmentKey = config.key;
+
+    assert(`[ember-metrics] You must pass a valid \`key\` to the ${this.toString()} adapter`, segmentKey);
+
+    delete config.key;
```

Comment:
Ah, yes. `config` [was being passed into `window.ga`](https://github.com/poteto/ember-metrics/blob/11a1bb61a350e74cc5d8c585bb595aa6637ad42d/addon/metrics-adapters/google-analytics.js#L41), so removing the `id` prop on the object was necessary to prevent errors being thrown by analytics.js. In this adapter, it does not appear that the Segment library takes options when being initialized, so deleting it seems unnecessary